From 9e41bcc34d7df92d036d48d2667bc50259c818aa Mon Sep 17 00:00:00 2001
From: Yury Norov <ynorov@caviumnetworks.com>
Date: Fri, 26 Oct 2018 08:30:52 +0200
Subject: [PATCH 0363/1051] pko: fixe Maximum Credit Limit for the MAX FIFO

This patch fixes Maximum Credit Limit for the MAX FIFO backed by SDP
Packet Output which different from the BGX.

Reported-by: Slawomir Rosek <slawomir.rosek@cavium.com>
CC: Aakash Sasidharan <Aakash.Sasidharan@cavium.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../net/ethernet/cavium/octeontx-83xx/pkopf_main.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c
index ba60d8700128..3b089e3aa310 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c
@@ -863,11 +863,19 @@ static int pko_mac_init(struct pkopf *pko, int mac_num, int mac_mode)
 	} else {
 		return -EINVAL;
 	}
-	reg = fifo | (skid << 5) | (0x0 << 15) | (0x1 << 16);
+	reg = fifo | (skid << 5) | (0x0 << 15) | (min_pad << 16);
 	pko_reg_write(pko, PKO_PF_MACX_CFG(mac_num), reg);
 
-	reg = bgx_txfifo_sz / 16; /* MAX_CRED_LIM */
-	pko_reg_write(pko, PKO_PF_MCI1_MAX_CREDX(mac_num), reg);
+	if (mac_num == SDP_MAC_NUM) { /* SDP MAC specific configuration */
+		/* SDP bug #28683 in mcbuggin  */
+		reg = 32; /* MAX_CRED_LIM */
+		dev_dbg(&pko->pdev->dev, "  write %016llx PKO_MCI1_MAX_CRED%d\n",
+			reg, mac_num);
+		pko_reg_write(pko, PKO_PF_MCI1_MAX_CREDX(mac_num), reg);
+	} else {  /* BGX MAC specific configuration */
+		reg = bgx_txfifo_sz / 16; /* MAX_CRED_LIM */
+		pko_reg_write(pko, PKO_PF_MCI1_MAX_CREDX(mac_num), reg);
+	}
 
 	reg = (rate << 3) | size;
 	pko_reg_write(pko, PKO_PF_PTGFX_CFG(ptgf), reg);
-- 
2.17.1

